src/disk_io_thread.cpp

```diff
@@ -3143,10 +3102,16 @@ namespace libtorrent
 	void disk_io_thread::submit_jobs()
 	{
 		std::unique_lock<std::mutex> l(m_job_mutex);
-		if (!m_queued_jobs.empty())
-			m_job_cond.notify_all();
-		if (!m_queued_hash_jobs.empty())
-			m_hash_job_cond.notify_all();
+		if (!m_generic_io_jobs.m_queued_jobs.empty())
```

Comment:
it's almost like this could be factored into a member function of the job queue.. but maybe not